Sources/GutenbergKit/Sources/EditorViewControllerDelegate.swift

```diff
@@ -20,9 +20,94 @@ public protocol EditorViewControllerDelegate: AnyObject {
     /// - warning: This is currently also called for the initial render, which
     /// is probably not how it should be in the production design.
     func editor(_ viewController: EditorViewController, didUpdateContentWithState state: EditorState)
+
+    func editor(_ viewController: EditorViewController, didRequestMediaFromSiteMediaLibrary config: OpenMediaLibrary)
 }
 
 public struct EditorState {
     /// Set to `true` if the editor has non-empty content.
     public var isEmpty = true
 }
+
+public struct OpenMediaLibrary: Decodable {
+    public let allowedTypes: [MediaType]
+    public let multiple: Bool
+    public let value: Value?
+
+    private enum CodingKeys: String, CodingKey {
+        case allowedTypes, multiple, value
+    }
+
+    public init(from decoder: Decoder) throws {
```

Comment:
Instead of a custom decoder, could `value` be an enum with arms for the different potential values?